tasks/copy_certs.yml

```diff
@@ -0,0 +1,42 @@
+- name: Copy certificate and private key to the host
+  block:
+    - name: Create a tempfile for the {{ item }} file on the host
+      tempfile:
+        state: file
+      register: __mssql_certificate_tempfile
+
+    - name: Copy the {{ item }} file to the host
+      copy:
+        src: "{{ item }}"
+        dest: "{{ __mssql_certificate_tempfile.path }}"
+        owner: mssql
+        group: mssql
+        mode: 0600
```

Comment:
Why use a tempfile?
Should there be a separate `cert_src` which is copied to `/etc/pki/tls/certs/` and `private_key_src` which is copied to `/etc/pki/tls/private/`?  If so, what should be the name of the files in `/etc/pki/tls/certs/` and `/etc/pki/tls/private/`?  does mssql expect a certain file name?  should the file name be something like `cert_src | basename` and `private_key_src | basename`?  Do you need to give the user the ability to specify the name of the file on the managed host e.g. by using `cert` and `private_key` as in https://linux-system-roles.github.io/documentation/tls_crypto_parameter_and_key_names.html ?

Reply:
mssql does not care about the names, I also see examples when people copied certs to different directories. I think mssql itself only needs to be pointed to the location of the cert and key. I think RHEL itself does some symlinks when you put the certs into this location. I think that the `cert and private_key` options are not needed, using the original names is enough.

Reply:
Customization is something that we can always add later. I think that what the users transferring from windows require now is as little customization as possible, hence I am trying to make it simpler.

Reply:
Ok - then you don't need the tempfile - you can use the `copy` module with the `src` is the `mssql_tls.cert`, and the dest is `/etc/pki/tls/certs/{{ mssql_tls.cert | basename }}`   https://docs.ansible.com/ansible/2.8/user_guide/playbooks_filters.html#id8 - you have to use `basename` because `mssql_tls.cert` could be a path and filename.

using `copy` makes it idempotent - it won't `copy` if the file already exists (unless you use `force` - but I'm not sure how to expose that to the user interface - a way for the user to specify force overwrite of existing files)

Reply:
I still don't understand why you are using a tempfile here.

Why not do this?

```yaml
    - name: Copy the {{ item }} file to the host
      copy:
        src: "{{ item }}"
        dest: /etc/pki/tls/{{ 'private' if item == mssql_tls.private_key else 'certs' }}/{{ __mssql_cert_tempfile.path | basename }}
        owner: mssql
        group: mssql
        mode: 0600
```

Reply:
I still don't understand what is the purpose of using `tempfile`.  Can you do this?
```yaml
- name: Copy the {{ item | basename }} TLS file to the appropriate location
  copy:
    src: "{{ item }}"
    dest: >-
      /etc/pki/tls/{{ 'private' if item == mssql_tls_private_key
      else 'certs' }}/{{ item | basename }}
    owner: mssql
    group: mssql
    mode: 0600
    force: "{{ mssql_tls_force }}"
```

Reply:
Oh, you mean tempfile at all. You recommended me using this in the previous PR that introduced the roles, these lines:
https://github.com/linux-system-roles/mssql/pull/2/files#diff-3d0ff1709ca48add100327bb2a468e6c508fb92a159c64c4f99ad1df89d9bddeR263
I though that this is a best practice rule. I can copy directly too.

Reply:
For the `mssql_input_sql_file` - you need to copy it somewhere on the managed host, and there isn't a "proper" location for it like `/etc/pki/tls/certs/` for TLS certificate files.  So the best way to handle it is to copy it to the remote host in a tempfile so that you can use it with `sqlcmd`.

Reply:
ooooh I see, thank you for the clarification, now it makes perfect sense. I will remove this tempfile block and just use the copy module tomorrow.

Reply:
Fixed now